draft-normington-jsonpath-latest.xml

```diff
@@ -202,24 +202,32 @@ root-selector = %x24               ; $ selects document root node
                   <name>Syntax</name>
                   <t>A dot child matcher has a key known as a dot child name or a single asterisk (<tt>*</tt>).
                   </t>
-                  <t>A dot child name corresponds to a key in a JSON object, without the surrounding double quotes (<tt>"</tt>).
+                  <t>A dot child name corresponds to a name in a JSON object. It can be unquoted or surrounded with quotes, either double (<tt>"</tt>) or single (<tt>'</tt>).
                   </t>
                   <sourcecode type="abnf">
 matcher = dot-child             ; see below for alternatives
 dot-child = %x2E dot-child-name / ; .&lt;dot-child-name&gt;
             %x2E %x2A             ; .*
-dot-child-name = TBD
+dot-child-name = union-child / ; "key" or 'key' (see below)
+                 unquoted-key
+unquoted-key = 1*(
+                  %x2D / ; -
+                  DIGIT / ; 0-9
```

Comment:
`DIGIT` and `ALFA` are defined in [appendix B.1 of ABNF RFC](https://tools.ietf.org/html/rfc5234#appendix-B.1), and they are widely used in other RFCs without re-defining. For example, [JSON RFC](https://tools.ietf.org/html/rfc8259#section-6) uses `DIGIT` rule as a part of JSON number definition.

Reply:
I agree that we could adopt that approach, but as we've discussed elsewhere, there isn't a formal way to pull in ABNF definitions from an RFC, so this introduces a formal weakness which I'd personally prefer to avoid.

Reply:
Okay, I see no problem in redefining some core ABNF rules that we use, I'll do that.

Reply:
Thanks!